### ISSUE TITLE
[feature]`sg scan -U` should not return exit code 0 in case errors were fixed

### ISSUE THREAD

**Issue by liiight** [ast-grep/ast-grep#1712]
# ⭐ Suggestion
Following #1708, I suggest `sg scan -U` not return 0 exit code if fixed were made. In fact, it should not affect the exit code that is return from running `sg scan` without the flag.

Using a non-zero exit code when errors are detected is common practice in other popular tools like [eslint](https://eslint.org/docs/latest/use/command-line-interface#exit-codes) or [ruff](https://docs.astral.sh/ruff/linter/#exit-codes) regardless if a `--fix` flag has been applied or not

# 💻 Use Cases
Frameworks like [pre-commit](https://pre-commit.com/index.html#new-hooks) rely on a non-zero exit code in case of errors and/or file modifications:
>The hook must exit nonzero on failure or modify files.

**Comment by HerringtonDarkholme** [comment]
I don't think a linter should exit with error if code is all fixed. 

See ruff https://docs.astral.sh/ruff/linter/#exit-codes

<img width="721" alt="image" src="https://github.com/user-attachments/assets/9168b1e3-18f3-48da-a8de-93b97ece38e2" />

The similar case applies to eslint, https://eslint.org/docs/latest/use/command-line-interface#--fix

<img width="866" alt="image" src="https://github.com/user-attachments/assets/9a0bdf04-bd48-4777-bf92-4da710437bfa" />

If you need error exit code, removing the `--fix` flag is a common convention.

**Comment by liiight** [comment]
first off, thank you for taking the time to respond, I appreciate it.
I admit you raise good points, and apparently I did misread the exit code situation with ruff and eslint.

Also, I realized that for my use case the exit code of ast-grep does not actually matter, since any changes it will make will be treated as a failure by pre-commit regardless, so for all intents and purposes, this is a moot point.

Thanks again for your time and your answer! I highly appreciate it 🙏
